### ISSUE TITLE
Invalid JS: Nested throw expressions

### ISSUE THREAD

**Issue by rightfold** [jashkenas/coffeescript#3789]
The following CoffeeScript code:

```
throw throw throw a
```

compiles to the following JavaScript code:

```
throw throw throw a;;;
```

which is invalid syntax in JavaScript.

Instead it should either compile to the following JavaScript code:

```
throw (function() {
    throw (function() {
        throw a;
    })();
})();
```

or be rejected by the CoffeeScript compiler.

Now this code is of course incredibly silly, but the CoffeeScript compiler should under no circumstance both accept the input and generate code with invalid syntax.

**Comment by jashkenas** [comment]
I don't think it should compile. That should be a syntax error.

**Comment by GeoffreyBooth** [comment]
See also #3709.

**Comment by GeoffreyBooth** [comment]
Fixed by #4664.
